compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -465,6 +465,43 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             );
         }
 
+        // Collect all possible candidates in the chain with same 'item_name'
+        let mut all_methods: Vec<super::probe::Candidate<'_>> = vec![];
+        if let Mode::MethodCall = mode && let SelfSource::MethodCall(mut rcvr_expr) = source {
```

Comment:
A trick for later: rustfmt today ignores the entire expression when it doesn't know what to do, and it doesn't know what to do with `if let` chains. You can temporarily change this to `if true {`, run rustfmt so that the body is properly formatted, and revert this line to what it is today. This will make it easier to ensure the code is properly formatted when running `./x.py fmt` without as much effort.